packages/compass-settings/src/components/modal.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from 'react';
+import { connect } from 'react-redux';
+import ipc from 'hadron-ipc';
+
+import {
+  Modal,
+  ModalTitle,
+  ModalFooter,
+  H3,
+  css,
+  spacing,
+  Button,
+} from '@mongodb-js/compass-components';
+
+import { fetchSettings } from '../stores/settings';
+
+import PrivacySettings from './settings/privacy';
+import Sidebar from './sidebar';
+
+type Settings = {
+  name: string;
+  component: React.ComponentType;
+};
+
+type SettingsModalProps = {
+  onInit: () => void;
+};
+
+const contentStyles = css({
+  display: 'flex',
+  minHeight: '400px',
+});
+
+const sideNavStyles = css({
+  width: '20%',
+});
+
+const settingsStyles = css({
+  width: '80%',
+  padding: spacing[2],
+});
+
+const settings: Settings[] = [{ name: 'Privacy', component: PrivacySettings }];
+
+export const SettingsModal: React.FunctionComponent<SettingsModalProps> = ({
+  onInit,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedSetting, setSelectedSettings] = useState(settings[0].name);
+
+  const SettingComponent =
+    settings.find((x) => x.name === selectedSetting)?.component ?? null;
+
+  useEffect(() => {
+    if (isOpen) {
+      onInit();
+    }
+  }, [onInit, isOpen]);
```

Comment:
What would be the reason to re-fetch them every time on modal open? Also the callback name sorta doesn't make sense in this case, so I'd suggest either updating the callback name or going back to doing init only once 🙂